translate/src/hooks/useTheme.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+
+function getSystemTheme(): string {
+  if (
+    window.matchMedia &&
+    window.matchMedia('(prefers-color-scheme: dark)').matches
+  ) {
+    return 'dark';
+  } else {
+    return 'light';
+  }
+}
+
+export function useTheme(initialTheme: string) {
+  useEffect(() => {
+    function applyTheme(newTheme: string) {
+      if (newTheme === 'system') {
+        newTheme = getSystemTheme();
+      }
+      document.body.classList.remove(
+        'dark-theme',
+        'light-theme',
+        'system-theme',
+      );
+      document.body.classList.add(`${newTheme}-theme`);
+    }
```

Comment:
That and getSystemTheme() is all that should remain in this hook.

Everything else should go back to Theme.tsx, because it's only needed there.